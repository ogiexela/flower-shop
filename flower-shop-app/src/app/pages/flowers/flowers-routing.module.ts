import { NgModule } from '@angular/core';
import {
  RouterModule,
  Routes,
} from '@angular/router';
import { FLOWER_ROUTE } from '../../constants/routes/flower.routes';
import { BuyPageComponent } from './buy-page/buy-page.component';
import { BuyPageModule } from './buy-page/buy-page.module';
import { SellPageComponent } from './sell-page/sell-page.component';
import { SellPageModule } from './sell-page/sell-page.module';

const routes: Routes = [
  {
    path: FLOWER_ROUTE.BUY,
    component: BuyPageComponent,
  },
  {
    path: FLOWER_ROUTE.SELL,
    component: SellPageComponent
  },
  {
    path: '',
    redirectTo: FLOWER_ROUTE.BUY,
    pathMatch: 'full',
  },
];

@NgModule({
  imports: [
    BuyPageModule,
    SellPageModule,
    RouterModule.forChild(routes),
  ],
  exports: [RouterModule]
})
export class FlowersRoutingModule { }
