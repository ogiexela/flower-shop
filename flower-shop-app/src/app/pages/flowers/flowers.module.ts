import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FlowerModule } from '../../components/flower/flower.module';
import { FlowersRoutingModule } from './flowers-routing.module';

@NgModule({
  imports: [
    CommonModule,
    FlowerModule,
    FlowersRoutingModule,
  ]
})
export class FlowersModule { }
