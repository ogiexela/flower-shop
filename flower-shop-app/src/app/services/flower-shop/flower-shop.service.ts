import { HttpClient } from '@angular/common/http';
import {
  Inject,
  Injectable,
} from '@angular/core';
import { Flower } from '../../models/flower.model';
import { Flowers } from '../../models/flowers.model';

@Injectable({
  providedIn: 'root'
})
export class FlowerShopService {

  private readonly API_URL = '/api';

  constructor(@Inject(HttpClient) private http: HttpClient) { }

  public getFlowers(page: number = 1, offset: number = 0, limit: number = 25) {
    return this.http.get<Flowers>(`${this.API_URL}/flower?limit=${limit}&offset=${offset}&page=${page}`);
  }

  public createFlower(flower: Flower) {
    return this.http.post<Flower>(`${this.API_URL}/flower`, flower);
  }
}
