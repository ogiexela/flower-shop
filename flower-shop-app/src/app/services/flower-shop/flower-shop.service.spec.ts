import {
  HttpClientTestingModule,
  HttpTestingController,
} from '@angular/common/http/testing';
import { TestBed } from '@angular/core/testing';
import { Flowers } from '../../models/flowers.model';
import { FlowerShopService } from './flower-shop.service';

describe('FlowerShopService', () => {
  let httpMock: HttpTestingController;

  beforeEach(() => TestBed.configureTestingModule({
    imports: [HttpClientTestingModule],
  }));

  it('should be created', () => {
    const service: FlowerShopService = TestBed.get(FlowerShopService);
    expect(service).toBeTruthy();
  });

  it('should call createFlower api', () => {
    const service: FlowerShopService = TestBed.get(FlowerShopService);
    httpMock = TestBed.get(HttpTestingController);

    service.createFlower({ name: 'original flower name' }).subscribe((createdFlower) => {
      expect(createdFlower).toMatchSnapshot('create flower api response');
    });



    const request = httpMock.expectOne('/api/flower');

    expect(request.request.method).toBe('POST');
    expect(request.request.body).toMatchSnapshot('create flower api call body');

    request.flush({ name: 'created flower name' });

    httpMock.verify();
  });

  it('should call getFlowers api with default args', () => {
    const service: FlowerShopService = TestBed.get(FlowerShopService);
    httpMock = TestBed.get(HttpTestingController);

    service.getFlowers().subscribe((flowers: Flowers) => {
      expect(flowers).toMatchSnapshot('get flowers api response');
    });


    const request = httpMock.expectOne(`/api/flower?limit=${25}&offset=${0}&page=${1}`);

    expect(request.request.method).toBe('GET');
    expect(request.request.params).toMatchSnapshot('get flower api call params');

    request.flush({
      count: 1,
      data: [ {name: 'dummy flower'}],
      page: 1,
      pageCount: 1,
      total: 1,
    });

    httpMock.verify();
  });
});
