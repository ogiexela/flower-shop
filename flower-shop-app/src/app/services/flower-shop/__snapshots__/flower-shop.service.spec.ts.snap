// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowerShopService should call createFlower api: create flower api call body 1`] = `
Object {
  "name": "original flower name",
}
`;

exports[`FlowerShopService should call createFlower api: create flower api response 1`] = `
Object {
  "name": "created flower name",
}
`;

exports[`FlowerShopService should call getFlowers api with default args: get flower api call params 1`] = `
HttpParams {
  "cloneFrom": null,
  "encoder": HttpUrlEncodingCodec {},
  "map": null,
  "updates": null,
}
`;

exports[`FlowerShopService should call getFlowers api with default args: get flowers api response 1`] = `
Object {
  "count": 1,
  "data": Array [
    Object {
      "name": "dummy flower",
    },
  ],
  "page": 1,
  "pageCount": 1,
  "total": 1,
}
`;
